src/sinks/aws_kinesis/firehose/config.rs

```diff
@@ -181,4 +187,13 @@ impl RetryLogic for KinesisRetryLogic {
         }
         is_retriable_error(error)
     }
+
+    fn should_retry_response(&self, response: &Self::Response) -> RetryAction {
+        if self.retry_partial && response.failure_count > 0 {
+            let msg = format!("partial error count {}", response.failure_count);
```

Comment:
It would be nice to include the error type and reason if we can pull that out of the response reasonably.